From 51ef4388dd704b11607e07a831fd4bf6662c020d Mon Sep 17 00:00:00 2001
From: Pardha Kesapragada <pkesapragada@habana.ai>
Date: Fri, 25 Sep 2020 07:38:29 +0300
Subject: [PATCH] [SW-21029] goya2: ppboot: changes to bringup pcie

Goya2 PCIE bring-up code has been modified with
correct settings needed in the PHY ip, to bring
up the link during PO. Just like Gaudi, the
udelay is replaced with wait for SRAM status
done.

Change-Id: Ib71b52cecaea5f225acff576e95f33c25fc513e5
Signed-off-by: Pardha Kesapragada <pkesapragada@habana.ai>
---

diff --git a/goya2_arc/common.c b/goya2_arc/common.c
index eafccf5..966dc69 100644
--- a/goya2_arc/common.c
+++ b/goya2_arc/common.c
@@ -60,17 +60,10 @@
 	 */
 	regmap_writel(1, mmPCIE_PHY_CR_PARA_SEL_0);
 	regmap_writel(1, mmPCIE_PHY_CR_PARA_SEL_1);
-	regmap_writel(1, mmPCIE_PHY_CR_PARA_SEL_2);
-	regmap_writel(1, mmPCIE_PHY_CR_PARA_SEL_3);
 
 	/* Select indirect access to CR PARA I/F instead of PRBI */
 	regmap_writel(1, mmPCIE_PHY_CR_PARA_CFG);
 
-	/*
-	 * Use PHY configuration from SoC online
-	 * instead of default values
-	 */
-	regmap_writel(1, mmPCIE_PHY_PHY_EXT_CTRL_SEL);
 	preboot_init_pcie_phy_fw_update();
 }
 
@@ -138,8 +131,9 @@
 	/* Disable the EQ REDO */
 	regmap_writel(0x00402801, mmPCIE_DBI_GEN3_RELATED_OFF);
 
-	/* delay before executing Synopsys W/A > 300us */
-	udelay(300);
+	while (!(regmap_readl(mmPCIE_PHY_PHYX_SRAM_INIT_DONE_0) &&
+		regmap_readl(mmPCIE_PHY_PHYX_SRAM_INIT_DONE_1)))
+		;
 
 	/* Execute the Synopsys W/A */
 	regmap_writel(0xF, mmPCIE_PHY_CR_PARA_CTRL);
